types/ChallengeBuilder.tsx

```diff
@@ -41,20 +42,31 @@ export const createChallenge = async (
   challenge_name: string,
   date: Date,
   description: string,
-  location?: Location.LocationObjectCoords,
+  location: LocationObject | null,
   image_id?: string,
   comment_id?: string,
 ): Promise<void> => {
   try {
     // Prepare the challenge data for Firestore
     const user: DBUser = await firestoreCtrl.getUser();
     console.log("createChallenge uid", user.uid);
+
+    if (location == null) {
+      console.log("location undefined");
+    }
+
+    // Convert the location object to a Firestore GeoPoint
+    let locationFirebase =
```

Comment:
change to accomodate for a optional location by wrapping it up in an if statement